base/shared/src/main/scala/scalaz/data/ilist.scala

```diff
@@ -160,7 +163,7 @@ object IListModule {
       IList.uncons(self)
 
     def append(that: IList[A]): IList[A] =
-      IList.reverse(that).foldLeft(self)((b, a) => IList.cons(a, b))
+      self.foldRight(that)(_ :: _)
 
     def :::(that: IList[A]): IList[A] =
       self.append(that)
```

Comment:
`that ::: self = that ++ self != self.append(that) = self ++ that`
Same thing with `++`. I guess both would eventually get caught by the tests :)

Reply:
`self.append(that)` is supposed to be `self ++ that`. Don't like the name?

Reply:
yes but `that ::: self` is not `self.append(that)`.

Reply:
Just add tests for `:::` and `++` and you'll see what I mean ;)

Reply:
Poop, fixed lol.